### PR TITLE
Feature: kubectl create ingress

### PR DESCRIPTION
**Pull request by amimof**
**What type of PR is this?**
/kind feature

**What this PR does / why we need it**:
Adds the ability to create ingress resources using `kubectl create ingress` command. This will speed up the development workflow by providing a quick and easy way of deploying *and* accessing apps. 

For example. As a developer I want to deploy my app to an existing staging cluster and be able to access it. In most cases I will be using `kubectl create serviceaccount`, `kubectl create service`, `kubectl create deployment`. But without an ingress I wont be able to access it, and `kubectl create ingress` does not exist currently.

**Which issue(s) this PR fixes**:

None

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:

```release-note
Adds `create ingress` command to `kubectl` 
```

### CODE REVIEW

File: staging/src/k8s.io/kubectl/pkg/cmd/create/create_ingress.go
Comment (soltysh) [inline staging/src/k8s.io/kubectl/pkg/cmd/create/create_ingress.go]: Long and example are usually defined outside for readability.

File: staging/src/k8s.io/kubectl/pkg/cmd/create/create_ingress.go
Comment (soltysh) [inline staging/src/k8s.io/kubectl/pkg/cmd/create/create_ingress.go]: I want to see `NewIngressOptions` method setting this.

File: staging/src/k8s.io/kubectl/pkg/cmd/create/create_ingress.go
Comment (soltysh) [inline staging/src/k8s.io/kubectl/pkg/cmd/create/create_ingress.go]: `IngressOptions`

File: staging/src/k8s.io/kubectl/pkg/cmd/create/create_ingress.go
Comment (soltysh) [inline staging/src/k8s.io/kubectl/pkg/cmd/create/create_ingress.go]: Defaults should be set in `NewIngressOptions`.

File: staging/src/k8s.io/kubectl/pkg/cmd/create/create_ingress.go
Comment (soltysh) [inline staging/src/k8s.io/kubectl/pkg/cmd/create/create_ingress.go]: Formatting is messed up here.

File: staging/src/k8s.io/kubectl/pkg/cmd/create/create_ingress.go
Comment (soltysh) [reply staging/src/k8s.io/kubectl/pkg/cmd/create/create_ingress.go]: Each flag should be wired with appropriate member of `IngressOptions` struct.

File: staging/src/k8s.io/kubectl/pkg/cmd/create/create_ingress.go
Comment (soltysh) [reply staging/src/k8s.io/kubectl/pkg/cmd/create/create_ingress.go]: You should be using values wired from `IngressOptions` struct.

File: staging/src/k8s.io/kubectl/pkg/cmd/create/create_ingress_test.go
Comment (soltysh) [inline staging/src/k8s.io/kubectl/pkg/cmd/create/create_ingress_test.go]: 2019 :stuck_out_tongue_winking_eye:

File: staging/src/k8s.io/kubectl/pkg/cmd/create/create_ingress_test.go
Comment (soltysh) [inline staging/src/k8s.io/kubectl/pkg/cmd/create/create_ingress_test.go]: That seems overly complicated. Have a look at other create command and their testing suites.

File: staging/src/k8s.io/kubectl/pkg/generate/versioned/ingress.go
Comment (soltysh) [inline staging/src/k8s.io/kubectl/pkg/generate/versioned/ingress.go]: Remove what's not used.

**Review by soltysh — CHANGES_REQUESTED** [review]

**Comment by soltysh** [comment]
Also, pls squash all into a single commit.

File: staging/src/k8s.io/kubectl/pkg/cmd/create/create_ingress.go
Comment (amimof) [reply staging/src/k8s.io/kubectl/pkg/cmd/create/create_ingress.go]: These are now created in two global cars

File: staging/src/k8s.io/kubectl/pkg/cmd/create/create_ingress.go
Comment (amimof) [reply staging/src/k8s.io/kubectl/pkg/cmd/create/create_ingress.go]: Looked at other commands and the standard seems to be `NewCreateIngressOptions`. Is this correct?

File: staging/src/k8s.io/kubectl/pkg/cmd/create/create_ingress.go
Comment (amimof) [reply staging/src/k8s.io/kubectl/pkg/cmd/create/create_ingress.go]: Renamed

File: staging/src/k8s.io/kubectl/pkg/cmd/create/create_ingress.go
Comment (amimof) [reply staging/src/k8s.io/kubectl/pkg/cmd/create/create_ingress.go]: Fixed this accordingly

File: staging/src/k8s.io/kubectl/pkg/generate/versioned/ingress.go
Comment (amimof) [reply staging/src/k8s.io/kubectl/pkg/generate/versioned/ingress.go]: Fixed

File: staging/src/k8s.io/kubectl/pkg/cmd/create/create_ingress_test.go
Comment (amimof) [reply staging/src/k8s.io/kubectl/pkg/cmd/create/create_ingress_test.go]: Looked at other commands as you suggested and got this sorted out

File: staging/src/k8s.io/kubectl/pkg/cmd/create/create_ingress.go
Comment (amimof) [reply staging/src/k8s.io/kubectl/pkg/cmd/create/create_ingress.go]: Fixed

File: staging/src/k8s.io/kubectl/pkg/cmd/create/create_ingress.go
Comment (amimof) [reply staging/src/k8s.io/kubectl/pkg/cmd/create/create_ingress.go]: Fixed

File: staging/src/k8s.io/kubectl/pkg/cmd/create/create_ingress_test.go
Comment (amimof) [reply staging/src/k8s.io/kubectl/pkg/cmd/create/create_ingress_test.go]: Fixed

File: staging/src/k8s.io/kubectl/pkg/cmd/create/create_ingress.go
Comment (soltysh) [inline staging/src/k8s.io/kubectl/pkg/cmd/create/create_ingress.go]: Use networking group here, extensions is already deprecated.

File: staging/src/k8s.io/kubectl/pkg/generate/versioned/generator.go
Comment (soltysh) [inline staging/src/k8s.io/kubectl/pkg/generate/versioned/generator.go]: Not needed, drop it.

File: staging/src/k8s.io/kubectl/pkg/generate/versioned/ingress.go
Comment (soltysh) [inline staging/src/k8s.io/kubectl/pkg/generate/versioned/ingress.go]: Not needed, drop it.

File: staging/src/k8s.io/kubectl/pkg/generate/versioned/ingress_test.go
Comment (soltysh) [inline staging/src/k8s.io/kubectl/pkg/generate/versioned/ingress_test.go]: Not needed, drop it.

**Review by soltysh — CHANGES_REQUESTED** [review]

File: staging/src/k8s.io/kubectl/pkg/cmd/create/create_ingress.go
Comment (amimof) [reply staging/src/k8s.io/kubectl/pkg/cmd/create/create_ingress.go]: Fixed

File: staging/src/k8s.io/kubectl/pkg/generate/versioned/generator.go
Comment (amimof) [reply staging/src/k8s.io/kubectl/pkg/generate/versioned/generator.go]: Fixed

File: staging/src/k8s.io/kubectl/pkg/generate/versioned/ingress.go
Comment (amimof) [reply staging/src/k8s.io/kubectl/pkg/generate/versioned/ingress.go]: I don't understand 😕Dropping the first line would break the comment?

**Review by soltysh — APPROVED** [review]
/lgtm
/approve
/ok-to-test

**Comment by amimof** [comment]
@soltysh @seans3 Alla tests have passed, just need lgtm label. Can you please check? Thanks

File: staging/src/k8s.io/kubectl/pkg/cmd/create/create_ingress.go
Comment (soltysh) [inline staging/src/k8s.io/kubectl/pkg/cmd/create/create_ingress.go:22]: Please split imports into following sections:
1. builtin go
2. 3-rd party
3. k8s
See other files, for example.

File: staging/src/k8s.io/kubectl/pkg/cmd/create/create_ingress.go
Comment (soltysh) [inline staging/src/k8s.io/kubectl/pkg/cmd/create/create_ingress.go:2]: 2020

File: staging/src/k8s.io/kubectl/pkg/cmd/create/create_ingress.go
Comment (soltysh) [inline staging/src/k8s.io/kubectl/pkg/cmd/create/create_ingress.go:29]: `networkingv1beta1client` is preferred alias

Comment (soltysh) [reply staging/src/k8s.io/kubectl/pkg/cmd/create/create_ingress.go:29]: Also there's `networking/v1` now, why not using that?

File: staging/src/k8s.io/kubectl/pkg/cmd/create/create_ingress.go
Comment (soltysh) [inline staging/src/k8s.io/kubectl/pkg/cmd/create/create_ingress.go:161]: You don't need to repeat since you have those marked required in flags.

File: staging/src/k8s.io/kubectl/pkg/generate/versioned/ingress.go
Comment (soltysh) [reply staging/src/k8s.io/kubectl/pkg/generate/versioned/ingress.go:1]: This file is not needed, since you have dedicated method in create ingress.

**Review by soltysh — CHANGES_REQUESTED** [review]

**Comment by rikatz** [comment]
Hello @amimof 

We've returned the discussion of kubectl create ingress due to issue #93267 and I've been working in this in the PR https://github.com/kubernetes/kubernetes/pull/94327

First of all, I'm sorry I didn't saw your PR before starting mine :/

Anyway we got some discussions in sig-network mailing list of what should be the command syntax, and I've developed with that in mind.

So I want to know if you want to move forward with your PR, if we should just close it or otherwise if you want to sync both works (mine and yours) you can ping me in Slack :)

Thank you!

File: go.mod
Comment (soltysh) [inline go.mod:25]: You should not modify any of these, not necessary in your PR.

**Review by soltysh — CHANGES_REQUESTED** [review]
Drop go.mod related changes and this can merge

**Comment by amimof** [comment]
After discussion with @rikatz we decided to merge work done in both PR's, this and https://github.com/kubernetes/kubernetes/pull/94327

**Review by soltysh — APPROVED** [review]
/lgtm

**Pull request merged**
